storagemarket/impl/providerstates/provider_states.go

```diff
@@ -311,7 +311,12 @@ func HandoffDeal(ctx fsm.Context, environment ProviderDealEnvironment, deal stor
 		carFilePath = string(file.OsPath())
 
 		// Hand the deal off to the process that adds it to a sector
+		log.Infow("handing off deal to sealing subsystem", "pieceCid", deal.Proposal.PieceCID, "proposalCid", deal.ProposalCid)
 		packingInfo, err = handoffDeal(ctx.Context(), environment, deal, file, uint64(file.Size()))
+		if err := file.Close(); err != nil {
+			log.Errorw("failed to close imported CAR file", "pieceCid", deal.Proposal.PieceCID, "proposalCid", deal.ProposalCid, "err", err)
```

Comment:
@raulk @nonsense This is concerning. We never closed the file stream we were handing off to the Miner.